src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1063,35 +1074,23 @@ void CouplingScheme::computeTimeStep(real &dt)
       if (mesh1.m_velX == nullptr || mesh1.m_velY == nullptr)
       {
          meshVel1 = false;
-         SLIC_WARNING( "CouplingScheme::computeTimeStep(): please register nodal "  << 
-                       "velocities for mesh id, " << m_meshId1 << ", in order to " <<
```

Comment:
Would it make sense to return an invalid value for the computed time step if it is not being calculated? Does this nullptr check occur on processors with no nodes?

Reply:
The idea is that the 'dt' input argument is the actual dt used by the host-code for the current cycle. So in the case where Tribol doesn't actually compute a timestep (e.g. null velocity pointers), then we simply don't modify the existing dt. To your point, in case the host-code doesn't actually pass their current timestep, I could return a very large (i.e. infeasible) timestep value that a host-code may see, and decide not to use. Thoughts?

Reply:
The nullptr check would occur on processors with zero-element meshes (e.g. no nodes). We simply wouldn't manipulate the dt input argument.

Reply:
> 

If someone is using the timestep vote, they are probably just taking a minimum of the returned timestep, so they probably wouldn't notice.

I can think of two ways to handle this:
1) Have two different calls, one that returns the timestep vote and one that doesn't.  Only give the warning if the former is called.  Note that you need to make sure the warning is not emitted if the velocity is nullptr on a processor with no nodes.
2) Return a negative value for the timestep vote if the information provided is incomplete.  You would probably have to keep the invalid flag separate from the timestep estimate and do a separate reduction.  You would also have to make sure a processor with no nodes doesn't return invalid.

Reply:
I'll let you think about that for a future TODO.

Reply:
I think for now the single call is fine. I removed the warning message, so we tolerate null-velocity pointers. Null-pointers would arise from a user not registering them, in which case the dt input argument is left unchanged, and the other if the pointers are registered, but happen to be null. Also in that case the dt input argument is unchanged. Rather than not changing the dt, which should be set at the current host-code's cycle dt, we could return a negative value allowing the host-code to query if the timestep vote was successful. Alternatively, while we always attempt to compute a tribol timestep vote, we could have the user set a Tribol option to compute it or not. On-rank a host-code could opt out of computing it for ranks with zero contact faces. Thoughts?

Reply:
I like returning the negative timestep if failed (assuming that this doesn't get set for a null ptr from an empty processor).